File: benchmark/nsieve/nsieve.lua
local function nsieve(limit)
  local flags = {}
  for i = 0, limit do
    flags[i] = 1
  end

  local p = 2
  while p * p <= limit do
    if flags[p] == 1 then
      local m = p * p
      while m <= limit do
        flags[m] = 0
        m = m + p
      end
    end
    p = p + 1
  end

  local count = 0
  for n = 2, limit do
    if flags[n] == 1 then
      count = count + 1
    end
  end
  return count
end

local total = 0
for rep = 1, 1 do
  total = total + nsieve(500000)
end
print(string.format("%d", total))

File: benchmark/nsieve/nsieve.php
<?php

function nsieve(int $limit): int
{
    $flags = array_fill(0, $limit + 1, 1);

    $p = 2;
    while ($p * $p <= $limit) {
        if ($flags[$p] == 1) {
            $m = $p * $p;
            while ($m <= $limit) {
                $flags[$m] = 0;
                $m += $p;
            }
        }
        $p += 1;
    }

    $count = 0;
    for ($n = 2; $n <= $limit; $n += 1) {
        if ($flags[$n] == 1) {
            $count += 1;
        }
    }
    return $count;
}

$total = 0;
for ($rep = 0; $rep < 60; $rep += 1) {
    $total += nsieve(500000);
}
echo $total, "\n";
